content-src/components/NewTabPage/NewTabPage.js

```diff
@@ -38,6 +38,24 @@ const NewTabPage = React.createClass({
     // have finished (in which case the "Welcome" dialog maybe be being shown
     // without any actual images).
     this.props.dispatch(actions.NotifyPerf("NEWTAB_RENDER"));
+
+    if (!this.props.isReady) {
+      this.props.dispatch(actions.NotifyUndesiredEvent({
+        event: "SHOW_LOADER",
+        page: PAGE_NAME
+      }));
+      this.loaderShownAt = Date.now();
+    }
+  },
+  componentDidUpdate(prevProps) {
+    if (this.props.isReady && this.loaderShownAt) {
+      this.props.dispatch(actions.NotifyUndesiredEvent({
+        event: "HIDE_LOADER",
+        page: PAGE_NAME,
+        timeShown: Date.now() - this.loaderShownAt
```

Comment:
Interesting.

Hey @dmose, iirc `AbsPerf.js` should work on the content side, correct?